Sources/GRPCHTTP2TransportNIOPosix/HTTP2ServerTransport+Posix.swift

```diff
@@ -62,6 +158,25 @@ extension HTTP2ServerTransport {
           }
         }
 
+      self.listeningAddressState.withLockedValue { state in
+        switch state {
+        case .idle(let idleState):
+          if let address = serverChannel.channel.localAddress {
+            idleState.listeningAddressPromise.succeed(address)
```

Comment:
Don't make outcalls while holding the lock. Generally speaking it's better to have functions on the `State` `enum` which return something to do. This makes it easier to separate state changes from side effects and it neatly encapsulates the logic of the state machine within the state type:

```swift
extension State { 
  mutating func bound(to address: GRPCHTTP2Core.SocketAddress) -> OnBind { ... }
  mutating func closed(error: ...) -> OnClosed { ... }

  var listeningAddress: EventLoopFuture<GRPCHTTP2Core.SocketAddress> { ... }
}

```